### PR TITLE
✨Story Ads: read amp-ad-exit 

### PR DESCRIPTION
**Pull request by calebcordry**
Temporary fix to allow reading of `amp-ad-exit` so that we can support ad networks that only include click tracking urls inside of `amp-ad-exit` config.

This should be deleted once these networks have first class support.

### CODE REVIEW

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js [hunk 1]
```diff
@@ -992,6 +1001,38 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
+      'The tag should contain exactly one <script> child.'
+    );
+    const child = children[0];
+    userAssert(
+      isJsonScriptTag(child),
+      'The amp-ad-exit config should ' +
+        'be inside a <script> tag with type="application/json"'
+    );
+    const config = assertConfig(parseJson(child.textContent));
+    const target = config['targets'][Object.keys(config['targets'])[0]];
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: protect the read by try-catch

**Comment by lannka** [comment]
Pls add a TODO and what is the criteria for deletion of the hack

File: extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js (hunk 1 above)
Comment (calebcordry) [reply extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js]: Done.

**Comment by calebcordry** [comment]
Created issue #24080 for tracking deletion of code, and added it to a few todos.

**Comment by calebcordry** [comment]
cc/ @sushan04

**Pull request merged**